### DISCUSSION TITLE
Windows XP, compiles but not runs

### DISCUSSION THREAD

**Discussion by L-u1** [magiblot/tvision#56] [General]
Hi,

thank you for the answers.

I have read tvision works on Windows XP in case of Borland C++ compiling.
Therefore I would like to know if tvision can be launched under XP but with MSVC?
At the moment XP says "The function GetCurrentConsoleFontEx not found in kernel32.dll"

Thank you

**Comment by magiblot** [comment]
I'm sorry, but no, MSVC builds do not work on Windows XP. There are several reasons why:

* `GetCurrentConsoleFontEx` and other dynamically-loaded functions are not provided in `kernel32.dll`. I once tried to avoid the dependency on `GetCurrentConsoleFontEx` but other errors came out which I was unable to solve.
* There is no support for [UTF-8 filesystem access](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/setlocale-wsetlocale?view=msvc-160#utf-8-support). So even if it built successfully, you would have trouble opening files with non-ascii names inside Turbo Vision applications (e.g. through the `TFileDialog`).

Cheers.

**Reply by paule32** [reply]
If you change the source tree, than you have to compile all neccassary files for the static library or the DLL Version, and your custom source tree. You can do this by using **CMAKE**.
You can decide, which build system you would use (Makefile, ninja). I prefer the `ninja` build system, because it is simple the fastest.
Ninja does only compile the changed files automatically invoke the headers, and build the archive .a files, and the .dll file.
If you compile the .DLL Version, try to add -fPIC to the g++ Compiler - it will be need.
I hope this Information's help a little bit, but comes with no guarantee.

Ciao

**Comment by magiblot** [comment]
Update: support for running in Windows XP was added in https://github.com/magiblot/tvision/commit/3154813d71a38b5c9772125e085c9dc69d41be0e. However, I have only been able to build XP-compatible binaries with GCC (not MSVC). See https://github.com/magiblot/tvision/pull/180#issuecomment-3220749121 for more details.
